shell_integration/windows/OCUtil/RemotePathChecker.cpp

```diff
@@ -90,15 +93,13 @@ void RemotePathChecker::workerThreadLoop()
                     // Remove any item from the cache
                     for (auto it = _cache.begin(); it != _cache.end() ; ) {
                         if (StringUtil::begins_with(it->first, responsePath)) {
+                            SHChangeNotify(SHCNE_UPDATEITEM, SHCNF_PATH | SHCNF_FLUSHNOWAIT, it->first.data(), NULL);
```

Comment:
The `SHCNF_FLUSHNOWAIT` should normally take care of that, is it possible that it wasn't there when you saw the deadlock?

I don't want to take the chance though, if you're not sure it might be better to get it out of the lock.